app/controllers/jobs_controller.rb

```diff
@@ -6,14 +6,19 @@ class JobsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
   def index
-    # @jobs = Job.geocoded
-    @markers = @jobs.map do |job|
-      {
-        lat: job.latitude,
-        long: job.longitude,
-        infoWindow: render_to_string(partial: 'info_window', locals: { job: job })
-      }
-    end
+    if params[:job].present?
+      @jobs = Job.where(category_id: params[:job][:query])
```

Comment:
Keep in mind that the `render_jobs` method also needs to run and might clash with this.